base/setup/reactos/drivepage.c

```diff
@@ -78,15 +311,50 @@ MoreOptDlgProc(HWND hwndDlg,
             {
                 case IDOK:
                 {
-                    GetDlgItemTextW(hwndDlg, IDC_PATH,
-                                    pSetupData->USetupData.InstallationDirectory,
-                                    ARRAYSIZE(pSetupData->USetupData.InstallationDirectory));
-                    EndDialog(hwndDlg, IDOK);
+                    HWND hEdit;
+                    BOOL bIsValid;
+                    WCHAR InstallDir[MAX_PATH];
+                    INT iItem;
+                    UINT uBldrLoc = CB_ERR;
+
+                    /*
+                     * Retrieve the installation path and verify its validity.
+                     * Check for the validity of the installation directory and
+                     * pop up an error if this is not the case.
+                     */
+                    hEdit = GetDlgItem(hDlg, IDC_PATH);
+                    bIsValid = (GetWindowTextLengthW(hEdit) < _countof(InstallDir)); // && IsValidInstallDirectory(InstallDir);
+                    GetWindowTextW(hEdit, InstallDir, _countof(InstallDir));
+                    bIsValid = bIsValid && IsValidInstallDirectory(InstallDir);
+
+                    if (!bIsValid)
+                    {
+                        // ERROR_DIRECTORY_NAME
```

Comment:
```suggestion
```
This is very obvious by the DisplayError()-call that follow at the next line.

Reply:
Except that this is the corresponding value from the `ERROR_NUMBER` enum (see `setup/lib/errorcode.h`), that I plan in the future to have a function to automatically map the corresponding strings to.
So it's added there as reference to the dev who's writing the code (me).